core/modules/replica/ExportServerConnection.cc

```diff
@@ -74,9 +74,9 @@ bool isErrorCode(boost::system::error_code const& ec, string const& scope) {
 bool readIntoBuffer(boost::asio::ip::tcp::socket& socket,
                     shared_ptr<ProtocolBuffer> const& ptr,
                     size_t bytes) {
-
-    // The buffer needs to be resized to accommodate
-    // the message.
+    // Make sure the buffer has enough space to accommodate the data
+    // of the message. Note, this call may throw an exception which is
+    // supposed to be caught by the method's caller.
```

Comment:
but you don't catch the exception. Should you?

Reply:
Not in this method. All exceptions are caught in the only client method:
```
template <class T>
bool readMessage(boost::asio::ip::tcp::socket& socket,
                 shared_ptr<ProtocolBuffer> const& ptr,
                 size_t bytes,
                 T& message) {
    try {
        if (readIntoBuffer(socket, ptr, bytes)) {
            ptr->parse(message, bytes);
            ...
```